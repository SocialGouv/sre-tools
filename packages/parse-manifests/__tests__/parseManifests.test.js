import parseManifests from "../src/index.js";

import fs from "fs";
import path from "path";

//console.log(parseManifests(fs.readFileSync("../sample.yml").toString()));

test("should parse manifests correctly", () => {
  const yaml = fs.readFileSync(path.join("sample.yml")).toString();
  expect(parseManifests(yaml)).toMatchSnapshot();
  expect(parseManifests(yaml).isProduction).toEqual(false);
});

test("PROD manifests correctly should be detected based on ingress rule", () => {
  const yaml = fs
    .readFileSync(path.join("sample.yml"))
    .toString()
    .replace(
      /(\s+)kubernetes\.io\/ingress\.class\:\snginx/g,
      `$1kubernetes.io/ingress.class: nginx
$1certmanager.k8s.io/cluster-issuer: letsencrypt-prod
`
    );
  expect(parseManifests(yaml).isProduction).toEqual(true);
});
