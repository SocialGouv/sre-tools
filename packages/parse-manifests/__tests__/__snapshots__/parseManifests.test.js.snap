// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse manifests correctly 1`] = `
Object {
  "app.github.com/repo": "company/app",
  "app.github.com/run": "1127909139",
  "deployments": Array [
    Object {
      "containers": Array [
        Object {
          "image": "ghcr.io/company/app/backend:1.1.0-alpha.6",
          "name": "backend",
        },
      ],
      "initContainers": Array [
        Object {
          "image": "ghcr.io/company/docker/wait-for-postgres:6.38.3",
          "name": "some-duplicate-image",
        },
        Object {
          "image": "ghcr.io/company/docker/wait-for-postgres:6.38.3",
          "name": "wait-for-postgres",
        },
      ],
      "name": "backend",
    },
    Object {
      "containers": Array [
        Object {
          "image": "ghcr.io/company/app/frontend-version1:1.1.0-alpha.6",
          "name": "version1",
        },
      ],
      "initContainers": Array [],
      "name": "version1",
    },
    Object {
      "containers": Array [
        Object {
          "image": "ghcr.io/company/app/frontend-version2:1.1.0-alpha.6",
          "name": "version2",
        },
      ],
      "initContainers": Array [],
      "name": "version2",
    },
  ],
  "hosts": Array [
    "backend-app-preprod.dev.company.com",
    "version1-app-preprod.dev.company.com",
    "version2-app-preprod.dev.company.com",
  ],
  "images": Array [
    "ghcr.io/company/app/backend:1.1.0-alpha.6",
    "ghcr.io/company/app/frontend-version1:1.1.0-alpha.6",
    "ghcr.io/company/app/frontend-version2:1.1.0-alpha.6",
    "ghcr.io/company/docker/wait-for-postgres:6.38.3",
  ],
  "isProduction": false,
  "manifests": Array [
    Object {
      "kind": "Deployment",
      "name": "backend",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "SealedSecret",
      "name": "backend-sealed-secret",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Service",
      "name": "backend",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Ingress",
      "name": "backend",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "SealedSecret",
      "name": "azure-app-volume",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "PersistentVolumeClaim",
      "name": "uploads",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "PersistentVolume",
      "name": "app-uploads",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Deployment",
      "name": "version1",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Service",
      "name": "version1",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Ingress",
      "name": "version1",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Deployment",
      "name": "version2",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Service",
      "name": "version2",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Ingress",
      "name": "version2",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "SealedSecret",
      "name": "azure-pg-user",
      "namespace": "app-preprod",
    },
    Object {
      "kind": "Ingress",
      "name": "xxx-redirect",
      "namespace": "xxx",
    },
  ],
  "namespace": "app-preprod",
  "redirects": Array [
    Object {
      "from": "www.some.url",
      "to": "https://target.fr$request_uri",
    },
    Object {
      "from": "www.some.alternative.url",
      "to": "https://target.fr$request_uri",
    },
  ],
}
`;
